lib/nodejs/lib/thrift/http_connection.js

```diff
@@ -214,6 +214,18 @@ HttpConnection.prototype.write = function(data) {
   var req = (self.https) ?
       https.request(self.nodeOptions, self.responseCallback) :
       http.request(self.nodeOptions, self.responseCallback);
+
+  //support for timeout
+  var timeout = self.nodeOptions.timeout;
+  if(timeout){
+    req.on('socket', function (socket) {
+        socket.setTimeout(timeout);  
+        socket.on('timeout', function() {
+            req.abort();
+        });
+    });
+  }
```

Comment:
@nsuke  One workaround could be to use a different option like "socketTimeout" and mention in documentation that this option is for node versions not having support for timeout.

Also, could you look patch submitted in this JIRA ticket: THRIFT-2968